### DISCUSSION TITLE
How to insert components into translation

### DISCUSSION THREAD

**Discussion by alphaCastor** [sveltekit-i18n/lib#120] [Q&A] [answered]
Hi there,
Thanks for your work, it is really easy to work with that library. 

Now I faced with one thing I don't know how to resolve.
I am working on an app and I have many texts like that:
<img width="840" alt="image" src="https://user-images.githubusercontent.com/93641159/236498876-27aca67a-0467-47b5-a093-cda1264cddf1.png">

As you can see here we have some components between raw text. Is it any way I can achieve it with library?

I would like to achieve something like that:

`en.json`

``` json
"key": "I highly recommend to charge ::link1 every day. On ::link2 step we will need to craft 12 pieces of food. It will require 12 days to get that amount of quartzes or you can buy it from Trading Post for gold. You can easily charge it moving to ::wp1 or ::wp2 and interacting with hero points:"
```
and in component:

```js
{t('charged_quarz', { link1: LinkComponent, link2: LinkComponent, wp1: WpComponent, wp2: WpComponent })
```

**Comment by jarda-svoboda** [comment]
Hi @alphaCastor.
As far as I know, the only way how to achieve that is to use [Svelte's `render`](https://svelte.dev/docs/server-side-component-api) method together with `@html` special tag like this:
```svelte
<script>
  import {browser} from '$app/environment';
</script>

{@html $t('key', { link: browser ? null : LinkComponent.render({/* component props */}).html })}
```

Note that the nested component won't be reactive (changing the component props won't re-render the component).

Hope that helps..

[a comment by an automated account is not shown]
